tests/virt_autotest/sev_snp_guest_verification.pm

```diff
@@ -0,0 +1,1270 @@
+#   - Removes incompatible features (TPM, SMM, memoryBacking)
+#   - Verifies SEV-SNP activation in guest dmesg
+#   - Runs attestation verification when supported
+# - Collects and uploads relevant logs
+#
+# Modes:
+# - Maintenance update: ENABLE_SEV_SNP_GUEST_VERIFICATION=1
+# - Unified guest installation: VIRT_SEV_SNP_GUEST_INSTALL=1
+#
+# Maintainer: QE-Virtualization <qe-virt@suse.de>
+
+package sev_snp_guest_verification;
+
+use base 'virt_feature_test_base';
+use strict;
+use warnings;
+use POSIX 'strftime';
+use File::Basename;
+use testapi qw(:DEFAULT);
+use serial_terminal qw(select_serial_terminal);
+use IPC::Run;
+use utils;
+use virt_utils;
+use virt_autotest::common;
+use virt_autotest::utils;
+use version_utils qw(is_sle package_version_cmp is_tumbleweed is_alp);
+use Utils::Architectures;
+use bmwqemu;
+use Utils::Logging;
+use package_utils;
+use bootloader_setup qw(add_grub_cmdline_settings grub_mkconfig);
+use power_action_utils 'power_action';
+use utils 'zypper_call';
+
+# Define constants for SNP verification
+use constant {
+    SNP_HOST_TOOLS => ['snphost', 'sevctl', 'snpguest'],
+    SNP_GUEST_TOOLS => ['snpguest'],
+    SNP_MIN_KERNEL_VER => '5.19.0',
+
+    # Log directory for collecting all test artifacts
+    LOG_DIR => '/tmp/sev_snp_test_logs'
+};
+
+###########################################
+#  SECTION 1: CORE TEST FLOW FUNCTIONS   #
+###########################################
+
+=head2 run_test
+
+  run_test($self)
+
+Main entry point for the test. This function:
+1. Determines the test context (unified guest install or maintenance update)
+2. Checks SEV-SNP support on the host
+3. Checks SEV-SNP support on each guest
+4. Records overall test status
+
+Supported SLES versions: 15-SP6, 15-SP7, and SLES 16+
+
+=cut
+
+sub run_test {
+    my $self = shift;
+
+    record_info('SEV-SNP Test Started', 'SEV-SNP verification test started');
+
+    # Determine test context
+    my $is_unified_guest_install = get_var("VIRT_UNIFIED_GUEST_INSTALL", 0) || get_var("VIRT_SEV_SNP_GUEST_INSTALL", 0);
+    my $is_maintenance_update_mode = get_var("ENABLE_SEV_SNP_GUEST_VERIFICATION", 1);
+
+    # Log test context based on mode
+    $self->_log_test_context($is_unified_guest_install, $is_maintenance_update_mode);
+
+    # Check SEV-SNP on host
+    $self->check_sev_snp_on_host;
+
+    # Check SEV-SNP on each guest
+    foreach my $guest (keys %virt_autotest::common::guests) {
+        $self->check_sev_snp_on_guest(guest_name => $guest);
+    }
+
+    record_info('SEV-SNP Test Completed', 'SEV-SNP verification test completed successfully');
+
+    return $self;
+}
+
+# Helper method to log test context
+sub _log_test_context {
+    my ($self, $is_unified_guest_install, $is_maintenance_update_mode) = @_;
+
+    # For unified guest installation context
+    if ($is_unified_guest_install) {
+        record_info('Test Context', 'Running in unified guest installation mode with SEV-SNP');
```

Comment:
How about telling explicitly if it is for development product test or MU test? Like ` record_info('Test Context -- Developing product test', 'Running in unified guest installation mode with SEV-SNP');`